code/go/0chain.net/miner/protocol_block.go

```diff
@@ -873,10 +873,12 @@ func (mc *Chain) generateBlock(ctx context.Context, b *block.Block,
 		return ErrLFBClientStateNil
 	}
 
-	b.Txns = make([]*transaction.Transaction, 0, mc.BlockSize())
+	b.Txns = make([]*transaction.Transaction, 0, 100)
+
+	initialBlockSize := mc.ChainConfig.MaxBlockCost() / mc.ChainConfig.TxnTransferCost()
```

Comment:
The cost of `transfer` transaction is not the average cost of all kinds of transactions. So let's not use it to calculate the block size. The initialBlockSize could be the same as `b.Txns` , it will be extended when the capacity is reached.